sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubIOTest.java

```diff
@@ -140,6 +149,66 @@ public void testNullSubscription() {
     assertNotNull(DisplayData.from(read));
   }
 
+  /**
+   * {@link PipelineOptions} with a ValueProvider.
+   */
+  public interface RuntimeSubscriptionOptions extends PipelineOptions {
+    ValueProvider<String> getValue();
+    void setValue(ValueProvider<String> provider);
+  }
+
+  @Test
+  public void testValueProviderSubscription() {
+    RuntimeSubscriptionOptions options =
+        PipelineOptionsFactory.create().as(RuntimeSubscriptionOptions.class);
+    options.setValue(StaticValueProvider.of("projects/project/subscriptions/subscription"));
+    ValueProvider<String> subscription = options.getValue();
+    Read<String> pubsubRead = PubsubIO.readStrings().fromSubscription(subscription);
+    Pipeline.create().apply(pubsubRead);
+    assertThat(pubsubRead.getSubscriptionProvider(), not(nullValue()));
```

Comment:
Lines 168-172 seem unnecessary - what could we possibly get wrong to make those lines fail?

Reply:
I figure it's a reasonable sanity check; there is some level of indirection in translating the actual subscription string to the "SubscriptionPath"